test/runners.jl

```diff
@@ -123,6 +123,32 @@ end
         end
     end
 
+    # Test that we get no warnings when compiling without linking and when building a shared lib with clang
+    @testset "Clang - $(platform)" for platform in platforms
+        mktempdir() do dir
+            is_broken = platform in [Platform("armv7l", "linux"; libc="musl"), Platform("armv6l", "linux"; libc="musl"),#https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/262
+                                     Platform("i686", "windows"), Platform("x86_64", "windows")]#https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/248
+            ur = preferred_runner()(dir; platform=platform)
+            iobuff = IOBuffer()
+            test_c = """
+            int test(void) {
+                return 0;
+            }
+            """
+            test_script = """
+            set -e
+            echo '$(test_c)' > test.c
+            clang -Werror -c test.c
+            clang -Werror -shared test.c -o test.\${dlext}
+            """
+            cmd = `/bin/bash -c "$(test_script)"`
+            @test run(ur, cmd, iobuff; tee_stream=devnull) broken=is_broken
+            seekstart(iobuff)
+            output = String(read(iobuff))
+            @test (occursin("error", output) || occursin("warning", output)) skip=is_broken
```

Comment:
I'm quite confused by this test to be honest.  I don't expect to see anything in the output, I'm having a hard time to understand what this is supposed to be doing,

Reply:
Right, `output` actually contains also the input string, so "error" is _always_ there.  This test as is isn't very useful.  Not sure there is anything worth checking in the output, the above `@test` may be enough for this case.

Reply:
Yeah, I got confused about that at first because I didn't expect that. But I guess the first test is already enough due to -Werror